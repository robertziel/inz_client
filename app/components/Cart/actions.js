import { ADD_TO_CART, ADD_ONE, REMOVE_ONE, CLEAR_CART } from './constants';

export function addToCart(item) {
  return {
    type: ADD_TO_CART,
    item,
  };
}

export function addOneMore(item) {
  return {
    type: ADD_ONE,
    item,
  };
}

export function substractOneMore(item) {
  return {
    type: REMOVE_ONE,
    item,
  };
}

export function clearCart() {
  return {
    type: CLEAR_CART,
  }
}
