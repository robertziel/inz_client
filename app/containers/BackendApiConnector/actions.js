import { SET_AUTHENTICATION_TOKEN } from './constants';

export function setAuthenticationToken(token) {
  return {
    type: SET_AUTHENTICATION_TOKEN,
    locale: token,
  };
}
