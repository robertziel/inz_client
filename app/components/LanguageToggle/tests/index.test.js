import React from 'react';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import { mount } from 'enzyme';

import { changeLocale } from 'containers/LanguageProvider/actions';
import LanguageProvider from 'containers/LanguageProvider';
import configureStore from 'configureStore';
import { translationMessages } from 'i18n';

import LocaleToggle, { mapDispatchToProps } from '../index';

describe('<LocaleToggle />', () => {
  let store;

  function mountWrapper() {
    return mount(
      <Provider store={store}>
        <LanguageProvider messages={translationMessages}>
          <LocaleToggle />
        </LanguageProvider>
      </Provider>,
    );
  }

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  describe('render', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = mountWrapper();
    });

    afterAll(() => {
      wrapper.unmount();
    });

    it('should match the snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('should present the default `en` english language option', () => {
      expect(wrapper.find('option[value="en"]')).not.toBeNull();
    });
  });

  describe('mapDispatchToProps', () => {
    describe('onLanguageToggle', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onLanguageToggle).toBeDefined();
      });

      it('should dispatch changeLocale when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        const locale = 'de';
        const evt = { target: { value: locale } };
        result.onLanguageToggle(evt);
        expect(dispatch).toHaveBeenCalledWith(changeLocale(locale));
      });
    });
  });
});
