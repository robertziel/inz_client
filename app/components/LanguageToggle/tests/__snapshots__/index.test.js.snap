// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocaleToggle /> render should match the snapshot 1`] = `"<div class=\\"Wrapper-sc-1k1j1l-0 YFgOt\\"><div class=\\"MuiInputBase-root\\"><div class=\\"MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInputBase-inputSelect\\" tabindex=\\"0\\" role=\\"button\\" aria-haspopup=\\"listbox\\">en</div><input type=\\"hidden\\" value=\\"en\\"><svg class=\\"MuiSvgIcon-root MuiSelect-icon\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M7 10l5 5 5-5z\\"></path></svg></div></div>"`;
